src/Analyzers/Core/Analyzers/RemoveUnnecessarySuppressions/AbstractRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,520 @@
+                    if (isDisable == nextPragmaIsDisable)
+                    {
+                        // Same pragma kind, increment the stack count
+                        matchingPragmaStackCount++;
+                    }
+                    else
+                    {
+                        // Found a pragma of opposite kind.
+                        if (matchingPragmaStackCount > 0)
+                        {
+                            // Not matching one for the input pragma, decrement stack count
+                            matchingPragmaStackCount--;
+                        }
+                        else
+                        {
+                            // Found the match.
+                            togglePragma = nextPragma;
+                            return true;
+                        }
+                    }
+                }
+
+                togglePragma = default;
+                return false;
+            }
+        }
+
+        private bool IsSupportedId(
+            SyntaxNode idNode,
+            [NotNullWhen(returnValue: true)] out string? id,
+            out bool isCompilerDiagnosticId)
+        {
+            id = idNode.ToString();
+
+            // Compiler diagnostic pragma suppressions allow specifying just the integral ID.
+            // For example:
+            //      "#pragma warning disable 0168" OR "#pragma warning disable 168"
+            // is equivalent to
+            //      "#pragma warning disable CS0168"
+            // We handle all the three supported formats for compiler diagnostic pragmas.
+
+            var idWithoutPrefix = id.StartsWith(CompilerErrorCodePrefix) && id.Length == CompilerErrorCodePrefix.Length + CompilerErrorCodeDigitCount
+                ? id.Substring(CompilerErrorCodePrefix.Length)
+                : id;
+
+            // ID without prefix should parse as an integer for compiler diagnostics.
+            if (int.TryParse(idWithoutPrefix, out var errorCode))
+            {
+                // Normalize the ID to always be in the format with prefix.
+                id = CompilerErrorCodePrefix + errorCode.ToString($"D{CompilerErrorCodeDigitCount}");
+                isCompilerDiagnosticId = true;
+                return _lazySupportedCompilerErrorCodes.Value.Contains(errorCode);
+            }
+
+            isCompilerDiagnosticId = false;
+            switch (id)
+            {
+                case IDEDiagnosticIds.RemoveUnnecessarySuppressionDiagnosticId:
+                    // Not supported as this would lead to recursion in computation.
+                    return false;
+
+                case "format":
+                case IDEDiagnosticIds.FormattingDiagnosticId:
+                    // Formatting analyzer is not supported as the analyzer does not seem to return suppressed IDE0055 diagnostics.
+                    return false;
+
+                default:
+                    return idWithoutPrefix == id;
+            }
+        }
+    }
+}
```

Comment:
i basically jumped here.  complexity level is too high for me in a tired state of mind.